OpenSim/Tools/CorrectionController.cpp

```diff
@@ -257,16 +255,26 @@ void CorrectionController::extendConnectToModel(Model& model)
     // create an actuator for each generalized coordinate in the model 
     // add these actuators to the model and set their indexes 
     const CoordinateSet& cs = _model->getCoordinateSet();
+    auto actuators = model.updComponentList<CoordinateActuator>();
+
     for(int i=0; i<cs.getSize(); i++) {
-        std::cout << " CorrectionController::extendConnectToModel(): " 
-                  <<  cs.get(i).getName()+"_corrector" << "  added " 
-                  << std::endl;
-        std::string name = cs.get(i).getName()+"_corrector";
-        CoordinateActuator *actuator = NULL;
-        if(_model->getForceSet().contains(name)){
-            actuator = (CoordinateActuator *)&_model->getForceSet().get(name);
+        const Coordinate& coord = cs[i];
+        std::string name = coord.getName() + "_corrector";
+
+        CoordinateActuator* actuator = nullptr;
```

Comment:
The code on L279–289 would use `actuator` defined on L278, but reusing the variable name makes the code difficult to read. Can you just use a `bool` (flag) here and change `if(!actuator)` on L273 to `if (!actuatorFound)`?

Reply:
I think I introduced a bug! The reason for the pointer is that if the actuator is found in the model it should be added to the working set of the Controller in L289, which should be outside the if/else block. Just happens that the test case does not have any correctors as part of the model (they are introduced and owned by the Controller itself).

Reply:
Ah ok, cool. Perhaps we should add a test case, or create an Issue to do it later?

Reply:
I made issue #2168 to track this. Thanks!